### ISSUE TITLE
something wrong in batchnorm between ctx.is_train in false and true

### ISSUE THREAD

**Issue by quanzongfeng** [apache/incubator-mxnet#3070]
I use lstm to train and test, use "cpu only"
I set BatchNormPara with:
eps: 0.001
momentum: 0.9
fix_gamma: false
use_global_state: false

In train phrase, all are good; but in test state, 
when use exec->Forward(false),  I get result with some data -nan(0x400000)
but use exec->Forward(true), all are good;

I check the code, and find that in batch norm phase,  when ctx.is_train == false, it just assign out with "moving_var" and "moving_mean";  
and i check the data in "out" after assign, some part is -nan(0x400000) when the data in "data" and "slope(gamma)" and "bias(beta)" are good.

why this?  why not calc var and mean when ctx.is_train == false?

**Comment by winstywang** [comment]
In bn we need to use the global stats from training phase in testing according to the original paper. 

Could you check that in the saved model the BN stats (namely, the auxiliary variables) are not nan?

Also, could you confirm that it works in GPU mode?

**Comment by quanzongfeng** [comment]
重新看了一下batch norm在lstm上应用的文章， 并且做了一下试验，发现了一个问题：
例如使用unrolled lstm， 序列长度为m， 对lstm的输入数据做batchnorm， 则对应m个batchnorm， batchnorm在初始化的时候可以指定gamma和beta 两个symbol，因此可以控制这m个batch norm都使用同一组gamma和beta参数，可是对于aux_states， 却仍然有m组， 因为目前在batch norm初始化的时候无法指定aux_states。 请问这个应该如何解决呢？

**Comment by phunterlau** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to reopen if this is still an active issue. Thanks!
